lms/djangoapps/instructor/views/api.py

```diff
@@ -2910,3 +2910,98 @@ def generate_certificate_exceptions(request, course_id, generate_for=None):
     }
 
     return JsonResponse(response_payload)
+
+
+USER_INDEX = 0
+NOTES_INDEX = 1
+
+
+@csrf_exempt
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@require_global_staff
+@require_POST
+def generate_bulk_certificate_exceptions(request, course_id):  # pylint: disable=invalid-name
+    """
+    Add Students to certificate white list from the uploaded csv file.
+    """
+    course_key = CourseKey.from_string(course_id)
+    students, warnings, errors, success = ([] for __ in range(4))
+    if 'students_list' in request.FILES:
+        try:
+            upload_file = request.FILES.get('students_list')
+            if upload_file.name.endswith('.csv'):
+                students = [row for row in csv.reader(upload_file.read().splitlines())]
+            else:
+                errors.append({
+                    'response': _('-- Make sure that the file you upload is in CSV format with no '
+                                  'extraneous characters or rows.')
+                })
+
+        except Exception:  # pylint: disable=broad-except
+            errors.append({
+                'response': _('-- Could not read uploaded file.')
```

Comment:
`--` is not suggested, I just added for better readability like a list or bullets etc. however we can remove this `--` character.